### PR TITLE
Small cleanup for object/factory code

### PR DESCRIPTION
**Pull request by ZehMatt**
Just QoL stuff, no functional changes.

### CODE REVIEW

File: src/openrct2/object/ObjectFactory.cpp [hunk 1]
```diff
@@ -251,9 +251,10 @@ namespace ObjectFactory
         }
     }
 
-    std::unique_ptr<Object> CreateObjectFromLegacyFile(IObjectRepository& objectRepository, const utf8* path, bool loadImages)
+    std::unique_ptr<Object> CreateObjectFromLegacyFile(
+        IObjectRepository& objectRepository, const std::string& path, bool loadImages)
```
Comment (duncanspumpkin) [inline src/openrct2/object/ObjectFactory.cpp:255]: shouldn't we be using u8string for this?

Comment (ZehMatt) [reply src/openrct2/object/ObjectFactory.cpp:255]: I suppose

**Review by duncanspumpkin — APPROVED** [review]
Both Get and Set both dont sound quite right for this function but a const set function is weird so think it makes sense being Get.

**Comment by IntelOrca** [comment]
I would rename it from Get***, change it to something like... UpdateRepositoryItemData. Anything starting with Get sounds like it should be stateless and without side effects.

**Comment by ZehMatt** [comment]
> I would rename it from Get***, change it to something like... UpdateRepositoryItemData. Anything starting with Get sounds like it should be stateless and without side effects.

Can't the same be said about UpdateRepositoryItemData? That sounds like it would mutate the object and not return data to the caller. I think I'll just have it actually return the struct but not in this PR, consider it to be a placeholder for now.

**Comment by IntelOrca** [comment]
> > I would rename it from Get***, change it to something like... UpdateRepositoryItemData. Anything starting with Get sounds like it should be stateless and without side effects.
> 
> Can't the same be said about UpdateRepositoryItemData? That sounds like it would mutate the object and not return data to the caller. I think I'll just have it actually return the struct but not in this PR, consider it to be a placeholder for now.

We are talking about this right?
```
void FootpathSurfaceObject::GetRepositoryItem(ObjectRepositoryItem& item) const
{
    item.FootpathSurfaceInfo.Flags = Flags;
}
```

It does not return data to the caller, so what it sounds like is correct and exactly what it is doing... mutating the object passed in and returning void.

**Comment by ZehMatt** [comment]
> > > I would rename it from Get***, change it to something like... UpdateRepositoryItemData. Anything starting with Get sounds like it should be stateless and without side effects.
> > 
> > 
> > Can't the same be said about UpdateRepositoryItemData? That sounds like it would mutate the object and not return data to the caller. I think I'll just have it actually return the struct but not in this PR, consider it to be a placeholder for now.
> 
> We are talking about this right?
> 
> ```
> void FootpathSurfaceObject::GetRepositoryItem(ObjectRepositoryItem& item) const
> {
>     item.FootpathSurfaceInfo.Flags = Flags;
> }
> ```
> 
> It does not return data to the caller, so what it sounds like is correct and exactly what it is doing... mutating the object passed in and returning void.
> 
> I don't think I would ever prefix a function name with `Get` unless it returned something.

That is exactly the function I'm talking about and it does return data by the reference passed as an argument, it does not return it by `return x;` sure, but it "gets" data for the caller which is evident by writing data into the passed reference. Your suggestion rather implies by the name that the object is updating state, reading `UpdateRepositoryItemData(item);` would definitely suggest to the reader that the object takes `item` and gets modified, that doesn't help readability in my opinion.

**Comment by duncanspumpkin** [comment]
What about Populate for the word.

**Comment by IntelOrca** [comment]
Or `CopyInfoTo(...)`

**Pull request closed without merging**